### PR TITLE
Yeyu/debug paralllel draft

### PR DESCRIPTION
**Pull request by yeyu-nvidia**
## What does this PR do?

**Type of change:** New feature

**Overview:** 
This PR implements kv cache in eagle training that significantly reduce memory consumption so that enables longer sequence length.
This PR also contains a prototype of parallel draft implementation. However, the training does not converge. I will leave it as a future work.

## Usage

```python
# Add a code snippet demonstrating how to use this
```

## Testing
Passed regression test.

## Before your PR is "*Ready for review*"

- **Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/CONTRIBUTING.md)** and your commits are signed.
- **Is this change backward compatible?**: Yes/No 
- **Did you write any new necessary tests?**: Yes/No
- **Did you add or update any necessary documentation?**: Yes/No
- **Did you update [Changelog](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/CHANGELOG.rst)?**: Yes/No 

## Summary by CodeRabbit

* **New Features**
  * Parallel-draft speculative decoding with configurable multi-step processing and learnable draft embeddings/hidden states.

* **Refactor**
  * Forwarding APIs now accept an optional inference context to propagate cache/offset info; per-step attention masking and sequence-offset tracking unified across base and draft flows.

* **Documentation**
  * Notes clarifying kv-cache and sequence-parallel limitations when using parallel drafting.

* **Chores**
  * Distributed data-consistency now warns and auto-aligns ranks by default (non-fatal).

### CODE REVIEW

**Comment by yeyu-nvidia** [comment]
Closing as stale. This branch dates to 2025 and has long since diverged from main (conflicting), and the EAGLE/parallel-draft work it explores has been superseded by the current speculative-decoding pipeline in main. Reopen if any of it is still wanted — nothing here is lost, the branch remains.
